dask/order.py

```diff
@@ -93,28 +94,28 @@ def ndependents(dependencies, dependents):
     >>> sorted(ndependents(dependencies, dependents).items())
     [('a', 3), ('b', 2), ('c', 1)]
     """
-    result = dict()
-
-    roots = [k for k, v in dependents.items() if not v]
-
-    result.update(dict((r, 1) for r in roots))
-
+    result = dict((k, 1) for k, v in dependents.items() if not v)
     leaves = [k for k, v in dependencies.items() if not v]
-
     for leaf in leaves:
-        _ndependents(leaf, result, dependencies, dependents)
-
+        _ndependents(leaf, result, dependents)
```

Comment:
I wonder if there is a way to do this once by dumping all of the leaves onto the stack and then iterating